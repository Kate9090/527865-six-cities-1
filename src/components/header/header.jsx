import React from 'react';
import PropTypes from 'prop-types';

import {connect} from 'react-redux';

import {Link} from 'react-router-dom';

import {getUser, getStatusAuthorization} from '../../reducer/user/selectors';


const Header = (props) => {
  const {user, checkAuthorization} = props;

  return <>
    <header className="header">
      <div className="container">
        <div className="header__wrapper">
          <div className="header__left">
            <Link to="/" className="header__logo-link" href="main.html">
              <img className="header__logo" src="/img/logo.svg" alt="6 cities logo" width="81" height="41" />
            </Link>
          </div>
          <nav className="header__nav">
            <ul className="header__nav-list">
              {!checkAuthorization ?
                <Link to="/login" className="header__login">Sign in</Link>
                : <>
                    <div className="header__avatar-wrapper user__avatar-wrapper"></div>
                    <Link to="/favorites" className="header__user-name user__name">{user.email}</Link>
                  </>}
            </ul>
          </nav>
        </div>
      </div>
    </header>
    </>;
};

Header.propTypes = {
  user: PropTypes.shape({
    id: PropTypes.number,
    email: PropTypes.string,
    name: PropTypes.string,
    avatarUrl: PropTypes.string,
    isPro: PropTypes.bool,
  }),
  checkAuthorization: PropTypes.bool.isRequired,
};

export {Header};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  user: getUser(state),
  checkAuthorization: getStatusAuthorization(state),
});

export default connect(
    mapStateToProps
)(Header);
