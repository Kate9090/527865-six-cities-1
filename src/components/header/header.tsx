import React from 'react';

import {connect} from 'react-redux';

import {Link} from 'react-router-dom';

import {UserData} from '../../types';
import {getUser, getStatusAuthorization} from '../../reducer/user/selectors';

interface Props {
  user: UserData,
  checkAuthorization: boolean,
}

const Header: React.FunctionComponent<Props> = (props) => {
  const {user, checkAuthorization} = props;

  return <>
    <header className="header">
      <div className="container">
        <div className="header__wrapper">
          <div className="header__left">
            <Link to="/" className="header__logo-link" href="main.html">
              <img className="header__logo" src="/img/logo.svg" alt="6 cities logo" width="81" height="41" />
            </Link>
          </div>
          <nav className="header__nav">
            <ul className="header__nav-list">
              {!checkAuthorization ?
                <Link to="/login" className="header__login">Sign in</Link>
                : <>
                    <div className="header__avatar-wrapper user__avatar-wrapper"></div>
                    <Link to="/favorites" className="header__user-name user__name">{user.email}</Link>
                  </>}
            </ul>
          </nav>
        </div>
      </div>
    </header>
    </>;
};

export {Header};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  user: getUser(state),
  checkAuthorization: getStatusAuthorization(state),
});

export default connect(
    mapStateToProps
)(Header);
