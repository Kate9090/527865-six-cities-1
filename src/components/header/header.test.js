import React from "react";
import renderer from 'react-test-renderer';

import {Header} from './header.jsx';
import {BrowserRouter} from 'react-router-dom';

it(`renders correctly header places part`, () => {
  const tree = renderer
    .create(<BrowserRouter><Header /></BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
