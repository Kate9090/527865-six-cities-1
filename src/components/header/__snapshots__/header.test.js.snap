// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly header places part 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          className="header__logo-link"
          href="main.html"
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="img/logo.svg"
            width="81"
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <a
            className="header__login"
            href="/login"
            onClick={[Function]}
          >
            Sign in
          </a>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;
